File: 59.螺旋矩阵-ii.cpp
/*
 * @lc app=leetcode.cn id=59 lang=cpp
 *
 * [59] 螺旋矩阵 II
 */

// @lc c ode=start
class Solution {
public:
    vector<vector<int>> generateMatrix(int n) {
        vector<vector<int>> result(n,vector<i nt>(n));
        int target=n*n;
        int number=1;
        int t=0,r=n-1,b= n-1,l= 0;
         while(number< =target){
             for(int i=l ;i<=r;i++) result[t][i]=number++;
            t++;
            for(int i=t ;i<=b;i++) result[i][r]=number++;
            r--;
            for(int i=r ;i>=l;i--) result[b][i]=number++;
            b--;
            for(int i=b ;i>=t;i--) result[i][l]=number++;
            l++;
        }
        return result;
    }
};
// @lc code=end

File: 59.螺旋矩阵-ii.py
#
# @lc app=leetcode.cn id=59 lang=python3
#
# [59] 螺旋矩阵 II
#

# @lc code=start
class Solution:
    def generateMatrix(self, n: int) -> List[List[int]]:
        result=[[0 for _ in range(n)] for _ in range(n)]
        t,r,b,l=0,n-1,n-1,0
        number,target=1,n**2
        while number<=target:
            for i in range(l,r+1):
                result[t][i]=number
                number+=1
            t+=1
            for i in range(t,b+1):
                result[i][r]=number
                number+=1
            r-=1
            for i in range(r,l-1,-1):
                result[b][i]=number
                number+=1
            b-=1
            for i in range(b,t-1,-1):
                result[i][l]=number
                number+=1
            l+=1
        return result
# @lc code=end

